test/tests/selftest/pbs_config_sched.py

```diff
@@ -46,97 +46,67 @@ class TestSchedConfig(TestSelf):
-        new_rr = self.get_a_setting(slines, 'round_robin')
-        new_nsk = self.get_a_setting(slines, 'node_sort_key')
-        new_jsk = self.get_a_setting(slines, 'job_sort_key')
-        self.assertEqual(new_rr, def_rr, "Change to unexpected setting")
-        self.assertEqual(new_nsk, def_nsk, "Change to unexpected setting")
-        self.assertEqual(new_jsk, ['"ncpus HIGH"'])
-        def_jsk = new_jsk
+        old_conf = new_conf
+        new_conf = set_and_get({'job_sort_key': '"ncpus HIGH"'})
+        diffs = cmp_dict(old_conf, new_conf)
+        self.assertEqual(diffs, [['job_sort_key', '"ncpus HIGH"']])
 
         # Test whether we can unset a value by setting to empty list
-        slines = self.set_and_get({'job_sort_key': []})
-        new_rr = self.get_a_setting(slines, 'round_robin')
-        new_nsk = self.get_a_setting(slines, 'node_sort_key')
-        new_jsk = self.get_a_setting(slines, 'job_sort_key')
-        self.assertEqual(new_rr, def_rr, "Change to unexpected setting")
-        self.assertEqual(new_nsk, def_nsk, "Change to unexpected setting")
-        self.assertEqual(new_jsk, [])
-        def_jsk = new_jsk
+        old_conf = new_conf
+        new_conf = set_and_get({'job_sort_key': []})
+        diffs = cmp_dict(old_conf, new_conf)
+        self.assertEqual(diffs, [['job_sort_key', '<missing>']])
+
+        # Test whether we can set multiple values for one setting
+        old_conf = new_conf
+        jsk = ['"job_priority HIGH"', '"ncpus HIGH"']
+        new_conf = set_and_get({'job_sort_key': jsk})
+        diffs = cmp_dict(old_conf, new_conf)
+        self.assertEqual(diffs, [['job_sort_key', jsk]])
 
-        # Test whether we can set multiple values
-        slines = self.set_and_get({'job_sort_key':
-                                   ['"job_priority HIGH"', '"ncpus HIGH"']})
-        new_rr = self.get_a_setting(slines, 'round_robin')
-        new_nsk = self.get_a_setting(slines, 'node_sort_key')
-        new_jsk = self.get_a_setting(slines, 'job_sort_key')
-        self.assertEqual(new_rr, def_rr, "Change to unexpected setting")
-        self.assertEqual(new_nsk, def_nsk, "Change to unexpected setting")
-        self.assertEqual(new_jsk, ['"job_priority HIGH"', '"ncpus HIGH"'])
+        # Test whether we can change multiple settings
+        old_conf = new_conf
+        jsk = ['"job_priority LOW"', '"mem LOW"']
+        new_conf = set_and_get({'primetime_prefix': 'xp_',
+                                'job_sort_key': jsk})
+        diffs = cmp_dict(old_conf, new_conf)
+        self.assertEqual(diffs, [['job_sort_key', jsk],
+                                 ['primetime_prefix', 'xp_']])
 
-        # Finally, check if the scheduler likes the final result
-        self.our_sched.set_sched_config(apply=True, validate=True)
+        # Finally, check if the scheduler likes the result
+        sched.set_sched_config(confs={}, apply=True, validate=True)
```

Comment:
Shouldn't this be confs=new_conf?

Reply:
Not in this case. Since we have been calling set_sched_config with the apply=True option, the config file gets written with the changes. Now, set_sched_config always starts with the contents of the current config file and merges in any confs= options. Since new_conf was just created by parsing the same file, we don't need to make any changes. So, passing an empty dictionary of changes is appropriate. Passing new_conf would also work.

Reply:
Makes sense, thanks